Bukkit/src/main/java/com/plotsquared/bukkit/player/BukkitPlayerManager.java

```diff
@@ -46,21 +47,26 @@
     private final PlotAreaManager plotAreaManager;
     private final EventDispatcher eventDispatcher;
     private final EconHandler econHandler;
+    private final PermissionHandler permissionHandler;
 
     @Inject public BukkitPlayerManager(@Nonnull final PlotAreaManager plotAreaManager,
                                        @Nonnull final EventDispatcher eventDispatcher,
-                                       @Nullable final EconHandler econHandler) {
+                                       @Nullable final EconHandler econHandler,
+                                       @Nonnull final PermissionHandler permissionHandler) {
         this.plotAreaManager = plotAreaManager;
         this.eventDispatcher = eventDispatcher;
         this.econHandler = econHandler;
+        this.permissionHandler = permissionHandler;
     }
 
     @Nonnull @Override public BukkitPlayer getPlayer(@Nonnull final Player object) {
+        if (!object.isOnline()) {
+            throw new NoSuchPlayerException(object.getUniqueId());
```

Comment:
Previously you could pass offline players and it would force load the fake entity via reflection. This is mostly here to catch those calls, as they should be using OfflinePlotPlayer instances instead, now that we support offline player permissions. Thus, you can ONLY get a PlotPlayer instance from an online player.

Reply:
But you agree to add a throws declaration on the PlayerManager class? Or is this behavior bukkit specific, in which case this could only be added to the javadoc?

Reply:
Javadoc and clarify that the player has to be *online*. Not that they could be offline (I don't think, at least). I do agree that it isn't clear enough.